### DISCUSSION TITLE
is stepmode supported in a rangeselector button? (todate|backwards)

### DISCUSSION THREAD

**Discussion by deltamelter** [dbuezas/lovelace-plotly-graph-card#468] [Q&A]
Noticed an example with rangeselector buttons, but on a chart showing days into the future it sets the range to display days in the past.
stepmode should fix this but doesn't seem to work in this card.
Is it implemented?  is it affected by anything else like general hours to show or offset?

Thanks

> [stepmode](https://plotly.com/javascript/reference/layout/xaxis/#layout-xaxis-rangeselector-buttons-items-button-stepmode)
> Parent: layout.xaxis.rangeselector.buttons[]
> Type: enumerated , one of ( "backward" | "todate" )
> Default: "backward"
> Sets the range update mode. If "backward", the range update shifts the start of range back "count" times "step" milliseconds. If "todate", the range update shifts the start of range back to the first timestamp from "count" times "step" milliseconds back. For example, with `step` set to "year" and `count` set to "1" the range update shifts the start of the range back to January 01 of the current year. Month and year "todate" are currently available only for the built-in (Gregorian) calendar.
>

**Comment by dbuezas** [comment]
Hi there, anything in plotlyjs should work. I am not aware of anything else in the card that would conflict with it.
If you can't make it work, you could also try out custom buttons: https://github.com/dbuezas/lovelace-plotly-graph-card/discussions/231#discussioncomment-4869001

**Reply by deltamelter** [reply]
I probably misunderstood what stepmode is supposed to do.
I want to have a button that focuses on future/forecasted data, eg next 3 days, next 7 days.
When I use a rangeselector button, it always show the previous 7 days.

**Reply by dbuezas** [reply]
Try custom buttons, you get a lot more flexibility!

**Reply by deltamelter** [reply]
I struggled at first, the buttons kept flipping the date order instead of just the data anchor
Got it working now though
